src/RedisSessionStateProvider/KeyGenerator.cs

```diff
@@ -13,17 +14,19 @@ internal class KeyGenerator
         public string DataKey { get; private set; }
         public string LockKey { get; private set; }
         public string InternalKey { get; private set; }
+        public string SessionDataType { get; private set; }
 
         private void GenerateKeys(string id, string app)
         {
             this.id = id;
-            DataKey = $"{{{app}_{id}}}_SessionStateItemCollection";
+            DataKey = $"{{{app}_{id}}}_{SessionDataType}";
```

Comment:
We should retain the "SessionStateItemCollection" portion of the key unless we need to eliminate it. What's your intention for the new `SessionDataType`? I assume to tag keys with a serializer identifier?

Reply:
This change has been reverted

Reply:
I think we should still include something like a "SerializerId" token in key names _in addition to "SessionStateItemCollection"_

Reply:
I added a suffix `_SessionStateItemCollection_MyCustomSerializer` but leave the value as `_SessionStateItemCollection` if the default serializer is used to provide backward compatibility.